src/pybind/mgr/nfs/cluster.py

```diff
@@ -89,7 +89,15 @@ def delete_config_obj(self):
                  f"{self.pool_ns}")
 
     @cluster_setter
-    def create_nfs_cluster(self, cluster_id, placement, virtual_ip):
+    def create_nfs_cluster(self,
+                           cluster_id: str,
+                           placement: Optional[str],
+                           virtual_ip: Optional[str],
+                           ingress: Optional[bool] = None):
+        if virtual_ip and not ingress:
+            raise orchestrator.OrchestratorError('virtual_ip can only be provided with ingress enabled')
+        if not virtual_ip and ingress:
+            raise orchestrator.OrchestratorError('ingress currently requires a virtual_ip')
```

Comment:
Let this code be within try block and raise `NFSInvalidOperation` exception instead of `OrchestratorError`